### PR TITLE
[ci] Split broker admin tests to unitgroup3

### PR DESCRIPTION
**Pull request by Shoothzj**
### Motivation
Currently, the broker-group run slowly, usually needed to run in an hour. Reduce the ci time

### Modifications
- change broker admin unit tests to group `broker-admin`
- add unit group3 to run `broker-admin` tests 

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 
  
- [x] `no-need-doc` 
  
  ci unit test changes, no need doc

### CODE REVIEW

File: .github/workflows/ci-unit-broker-broker-gp3.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+        with:
+          distribution: 'temurin'
+          java-version: 11
+
+      - name: build modules
+        if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+        run: mvn -B -ntp -q clean install -Pcore-modules,-main -DskipTests
+
+      - name: run unit test 'BROKER_GROUP_3'
```
Comment (nicoloboschi) [inline .github/workflows/ci-unit-broker-broker-gp3.yaml:78]: what about refactoring this file having a single yaml file for 'broker group' and using a matrix for the group variable ? just like there
https://github.com/apache/pulsar/blob/master/.github/workflows/ci-owasp-dependency-check.yaml#L36

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Comment by Shoothzj** [comment]
@lhotari @nicoloboschi Do you know if refactoring this actions to matrix have impacted on `/pulsarbot run-failure-checks`. I am not familiar with `pulsar-bot`. Thanks :)

**Comment by nicoloboschi** [comment]
I don't expect problems for the pulsarbot. GH renders to different jobs, just like copy-pasting the yaml file so we should be good

**Comment by Shoothzj** [comment]
@nicoloboschi PTAL again, thanks

File: .github/workflows/ci-unit-broker-broker-gp3.yaml (hunk 1 above)
Comment (Shoothzj) [reply .github/workflows/ci-unit-broker-broker-gp3.yaml]: fixed

File: .github/workflows/ci-unit-broker-broker-gp.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@
 # under the License.
 #
 
-name: CI - Unit - Brokers - Broker Group 1
+name: CI - Unit - Brokers - Broker Group
```
Comment (nicoloboschi) [inline .github/workflows/ci-unit-broker-broker-gp.yaml:20]: Can you add the matrix group in the name ?

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Comment by Shoothzj** [comment]
@nicoloboschi they are already in github actions. what's your expected behavior?
image

**Comment by nicoloboschi** [comment]
In this way we can easily differentiate them in this list https://github.com/apache/pulsar/actions

However it's not blocking for me

**Review by nicoloboschi — APPROVED** [review]

**Review by merlimat — APPROVED** [review]

**Pull request merged**